src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -3840,3 +4003,46 @@ class CallStatementSegment(BaseSegment):
             ),
         ),
     )
+
+
+@snowflake_dialect.segment(replace=True)
+class OrderByClauseSegment(BaseSegment):
+    """A `ORDER BY` clause like in `SELECT`."""
+
+    type = "orderby_clause"
+    match_grammar = StartsWith(
+        Sequence("ORDER", "BY"),
+        terminator=OneOf(
+            "LIMIT",
+            "HAVING",
+            "QUALIFY",
+            # For window functions
+            "WINDOW",
+            Ref("FrameClauseUnitGrammar"),
+            "SEPARATOR",
+            "MEASURES",
```

Comment:
This basically just adds `MEASURES` to the `ORDER BY` terminator list

Reply:
You can insert it like this:

```
    match_grammar = ansi_dialect.get_segment(
        "OrderByClauseSegment"
    ).match_grammar.copy()
    match_grammar.terminator = (
        match_grammar.terminator.copy(
            insert=["MEASURES"],
        )
    )
```

Reply:
Nice will use this in #2784 as well 😄

Reply:
FYI the insert expects a list of references so I had to use `Ref.keyword` to get this to work:
```python
    match_grammar = ansi_dialect.get_segment(
        "OrderByClauseSegment"
    ).match_grammar.copy()
    match_grammar.terminator = (
        match_grammar.terminator.copy(
            insert=[Ref.keyword("MEASURES")],
        )
    )
```

Reply:
Yeah did wonder if you'd have to do that.